src/shared/rpc/methods/validation.ts

```diff
@@ -0,0 +1,61 @@
+import { PaymentTypes } from '@btckit/types';
+import * as btc from '@scure/btc-signer';
+import * as yup from 'yup';
+
+import { networkModes } from '@shared/constants';
+import { isNumber, isUndefined } from '@shared/utils';
+
+export function validateRpcParams(
+  obj: unknown,
+  validator: yup.ObjectSchema<yup.Maybe<yup.AnyObject>>
+) {
+  try {
+    validator.validateSync(obj, { abortEarly: false });
+    return true;
+  } catch (e) {
+    return false;
+  }
+}
```

Comment:
Nice abstraction of the validation fn, but like Pete said, think a cleaner end API would be to create separate functions that use this, rather than passing it in as an arg when we need it:

One way might be with a higher order fn:

```ts
// returns (obj: unknown)=> boolean
const validateRpcSendMessageParams = createValidator(rpcSendMessageParamsSchema);

// or wrapper fn
function validateRpcSendMessageParams(obj: unknown) {
  return validateRpcParams(obj, rpcSendMessageParamsSchema);
}
```